olive/model.py

```diff
@@ -103,6 +110,51 @@ def validate_type(cls, v):
     def create_model(self):
         return REGISTRY[self.type.lower()](**self.config)
 
+def _select_execution_providers(model: OliveModel, inference_settings: Dict[str, Any], device: Device) -> List[str]:
```

Comment:
eps is only for OnnxModel. It would be better to use `model: ONNXModel`.

Reply:
Introduced ONNModelBase that works as a base class for both ONNXModel & DistributedOnnxModel